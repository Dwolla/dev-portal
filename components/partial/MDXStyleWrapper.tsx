import { css } from "@emotion/core";
import { breakUp } from "../breakpoints";
import {
  headingStyles,
  paragraphStyles,
  codeStyles,
  linkStyles,
  listStyles,
} from "../base/Typography";
import { imageStyles } from "../base/Image";

const MDXStyleWrapper = ({ children }: { children: any }) => (
  <div
    css={css`
      padding: 20px;

      @media (${breakUp("md")}) {
        padding: 20px 40px;
      }

      > h1,
      h2,
      h3,
      h4,
      h5,
      h6 {
        ${headingStyles.global.styles}
        code {
          ${codeStyles.styles}
          font-size: inherit;
        }
      }

      > h1 {
        ${headingStyles.h1.styles}
      }

      > h2 {
        ${headingStyles.h2.styles}
      }

      > h3 {
        ${headingStyles.h3.styles}
      }

      > h4 {
        ${headingStyles.h4.styles}
      }

      > h5 {
        ${headingStyles.h5.styles}
      }

      > p:not([class]),
      blockquote {
        ${paragraphStyles.styles}

        > code {
          ${codeStyles.styles}
        }

        > img {
          ${imageStyles.styles}
        }
      }

      p {
        code {
          ${codeStyles.styles}
        }
      }

      > table,
      th,
      td {
        code {
          ${codeStyles.styles}
        }
      }

      a:not([class]) {
        ${linkStyles.styles}
      }

      > ol,
      > ul {
        ${listStyles.styles}
        code {
          ${codeStyles.styles}
        }
      }

      > img {
        ${imageStyles.styles}
      }

      span {
        code {
          ${codeStyles.styles}
        }
      }
    `}
  >
    {children}
  </div>
);

export default MDXStyleWrapper;
