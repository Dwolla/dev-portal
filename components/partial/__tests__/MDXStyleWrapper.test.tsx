import renderer from "react-test-renderer";
import MDXStyleWrapper from "../MDXStyleWrapper";

test("MDXStyleWrapper", () => {
  const tree = renderer
    .create(
      <MDXStyleWrapper>
        <h1>This is wrapped in the MDX Style Wrapper</h1>
      </MDXStyleWrapper>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
