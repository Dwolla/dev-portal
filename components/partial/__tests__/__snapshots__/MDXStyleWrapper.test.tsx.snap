// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXStyleWrapper 1`] = `
<div
  className="css-1o9c7k-global-global-codeStyles-codeStyles-h1-h2-h3-h4-h5-baseTextStyles-baseTextStyles-paragraphStyles-paragraphStyles-codeStyles-codeStyles-imageStyles-imageStyles-codeStyles-codeStyles-linkStyles-linkStyles-baseTextStyles-baseTextStyles-listStyles-listStyles-codeStyles-codeStyles-imageStyles-imageStyles-MDXStyleWrapper-MDXStyleWrapper"
>
  <h1>
    This is wrapped in the MDX Style Wrapper
  </h1>
</div>
`;
